app/src/main/java/com/geeksville/mesh/ui/UsersFragment.kt

```diff
@@ -142,6 +103,49 @@ fun NodesScreen(
 
     val currentTimeMillis = rememberTimeTickWithLifecycle()
 
+    var displayIgnoreDialogForNode: NodeEntity? by remember { mutableStateOf(null) }
+    var displayRemoveDialogForNode: NodeEntity? by remember { mutableStateOf(null) }
+    if (displayIgnoreDialogForNode != null) {
+        val isIgnored = state.ignoreIncomingList.contains(displayIgnoreDialogForNode!!.num)
+        SimpleAlertDialog(
+            title = R.string.ignore,
+            text = stringResource(
+                id = if (isIgnored) R.string.ignore_remove else R.string.ignore_add,
+                displayIgnoreDialogForNode?.user?.longName.orEmpty()
+            ),
+            onConfirm = {
+                model.ignoreIncomingList =
+                    state.ignoreIncomingList.toMutableList().apply {
+                        if (isIgnored) {
+                            debug("removed '${displayIgnoreDialogForNode!!.num}' from ignore list")
+                            remove(displayIgnoreDialogForNode!!.num)
+                        } else {
+                            debug("added '${displayIgnoreDialogForNode!!.num}' to ignore list")
+                            add(displayIgnoreDialogForNode!!.num)
+                        }
+                    }
+                displayIgnoreDialogForNode = null
+            },
+            onDismiss = {
+                displayIgnoreDialogForNode = null
+            }
+        )
+    }
+    if (displayRemoveDialogForNode != null) {
+        SimpleAlertDialog(
+            title = R.string.remove,
+            text = R.string.remove_node_text,
+            onConfirm = {
+                displayIgnoreDialogForNode?.let {
+                    model.removeNode(it.num)
+                }
+                displayRemoveDialogForNode = null
+            },
+            onDismiss = {
+                displayRemoveDialogForNode = null
+            }
+        )
+    }
```

Comment:
can this block move inside the NodeMenu component?

Reply:
I was receiving crashes when this was contained in the menu.. I'll give it another shot.